hbase-shell/src/main/ruby/shell/commands/regioninfo.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Shell
+  module Commands
+    class Regioninfo < Command
```

Comment:
nit: Any chance we could include an optional flag to force it to treat the param as encoded / not encoded?

Reply:
How would that work? You pass name. We figure internally if you passed us an encoded name or a plain name. Dependent on what we think you passed us, then we set region specifier type accordingly.

Reply:
I added more to the example. Perhaps that helps explain how this thing is used?

Reply:
yeah that helps.